src/Skill/Progression.cpp

```diff
@@ -23,35 +23,37 @@ namespace CC
 //Leveling up adds one skillpoint to every skill
 	void Progression::levelUp()
 	{
-		level_++;
+		++level_;
+		for (int& i : skills_)
+			++i;
 	}
 
 	bool Progression::addXp(int xp)
 	{
 		xp_ += xp;
 		int neededXpToLevel = neededXpToLevelUp();
-		if (neededXpToLevel >= 0)
+		if (neededXpToLevel <= 0)
 		{
 			levelUp();
-			xp_ = neededXpToLevel;
+			xp_ = -neededXpToLevel;
```

Comment:
Did you mean to put the minus before the equal?

Reply:
yes

Reply:
If there was like 1 xp needed to level up but you got 30 xp : there is -29 xp needed so when you are the next level. Now you have one more level + 29 xp...

Reply:
Reread my question

Reply:
Reread my question

Reply:
I knew what you wanted to do. Please reread my question

Reply:
I knew what you wanted to do. Please reread my question more carefully

Reply:
no then

Reply:
Alright, I see what I wrongly interpreted